### DISCUSSION TITLE
How do I hide antd dependency's warning log???

### DISCUSSION THREAD

**Discussion by jkiss** [ant-design/ant-design#41647] [Q&A] [answered]
there some annoying log when I use antd@5.4.0 like below:

```
Warning: 'dateRender' is deprecated. Please use 'cellRender' instead.
```

I don't want to see this in developing or production, how do I do?

**Comment by zombieJ** [comment]
warning only show in dev env. You can spy on console.error to hide this.

**Reply by jkiss** [reply]
but it will break my own logic code, dependency's warning should show when developing antd NOT using antd, right?

**Comment by day-xue** [comment]
Yes, me too. u can use below the 5.4.0 version to try again

**Reply by jkiss** [reply]
I tried downgrade version but no work, maybe need delete `package-lock.json` and `node_modules` which cost a lot

**Comment by tunaon** [comment]
Yes, can use the 5.4.0 version 
i will this issue monitoring

**Comment by Emeshka** [comment]
The same issue on 5.4.0. I use RangePicker component in my code.
I think the dependency warning should not be shown to me since I don't use dateRender or monthCellRender, DatePicker from antd itself does.
```
Warning: 'dateRender' is deprecated. Please use 'cellRender' instead.
warning @ warning.js:6
call @ warning.js:20
warningOnce @ warning.js:25
InnerRangePicker @ RangePicker.js:619
renderWithHooks @ react-dom.development.js:16305
mountIndeterminateComponent @ react-dom.development.js:20074
beginWork @ react-dom.development.js:21587
beginWork$1 @ react-dom.development.js:27426
performUnitOfWork @ react-dom.development.js:26557
workLoopSync @ react-dom.development.js:26466
renderRootSync @ react-dom.development.js:26434
performConcurrentWorkOnRoot @ react-dom.development.js:25738
workLoop @ scheduler.development.js:266
flushWork @ scheduler.development.js:239
performWorkUntilDeadline @ scheduler.development.js:533

warning.js:6 Warning: 'monthCellRender' is deprecated. Please use 'cellRender' instead.
warning @ warning.js:6
call @ warning.js:20
warningOnce @ warning.js:25
InnerRangePicker @ RangePicker.js:620
renderWithHooks @ react-dom.development.js:16305
mountIndeterminateComponent @ react-dom.development.js:20074
beginWork @ react-dom.development.js:21587
beginWork$1 @ react-dom.development.js:27426
performUnitOfWork @ react-dom.development.js:26557
workLoopSync @ react-dom.development.js:26466
renderRootSync @ react-dom.development.js:26434
performConcurrentWorkOnRoot @ react-dom.development.js:25738
workLoop @ scheduler.development.js:266
flushWork @ scheduler.development.js:239
performWorkUntilDeadline @ scheduler.development.js:533
```

**Comment by zombieJ** [comment]
antd using `rc-util/lib/warning` for warning info. You can config webpack alias to change to an empty function. But we still recommend to handle warning info if upgrade antd version.

**Reply by jkiss** [reply]
but user can NOT handle warning that NOT created by themselves, because it emit from internal plugins which antd dependent :(

**Reply by jkiss** [reply]
![image](https://user-images.githubusercontent.com/3231326/231128488-1628a48d-0f30-4b8e-a9e1-d5a96f41a587.png)

maybe DatePicker update but RangePicker not... 🤔

**Comment by barisdevjs** [comment]
I have the same issue but it is not present on Live. It appears only dev mode.

**Reply by jkiss** [reply]
yes, a little annoying that shifting our `console.log` 😭

**Comment by zombieJ** [comment, marked as answer]
Added https://github.com/react-component/util/pull/437 . Now can use `preMessage` to filter info instead mock console directly.

**Reply by ThePiyushAggarwal** [reply]
I was using 5.4.0 and was getting this warning. I upgraded to 5.4.2 and I think because of your fix, I am no longer getting that warning.

**Reply by ekeijl** [reply]
For anyone who needs this, here's how to use it:

Install the `@rc-component/util` package, make sure you use the same version as your Ant Design version uses. See [antd package.json](https://github.com/ant-design/ant-design/blob/master/package.json#L154).

Import preMessage and call it in the root of your app (index.js, App.js, etc), pass it a custom callback where you check for the warning you want to suppress and return null:

```js
import { preMessage } from '@rc-component/util/es/warning';

preMessage((message) => {
  // customize to your needs
  if (message?.includes('List')) { 
     return null;
  }
});
```
